core/include/vecmem/utils/attributes.hpp

```diff
@@ -0,0 +1,19 @@
+/*
+ * VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2021-2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+#ifdef __has_cpp_attribute
+#if __has_cpp_attribute(nodiscard) >= 201603L
+#define VECMEM_NODISCARD [[nodiscard]]
+#else
+#define VECMEM_NODISCARD
+#endif
+#else
+#define VECMEM_NODISCARD
+#endif
```

Comment:
Hmm... Now that I checked this because of another reason...

Why did you bother with all of this here? We already require clients to use a minimum of C\+\+17 with their host compiler to be able to use the core functionality of this project. (That of memory resources.) So we can just use `[[nodiscard]]` directly. Really no need for macros here. :thinking: